canal/handler.go

```diff
@@ -17,6 +17,14 @@ type EventHandler interface {
 	OnGTID(header *replication.EventHeader, gtid mysql.GTIDSet) error
 	// OnPosSynced Use your own way to sync position. When force is true, sync position immediately.
 	OnPosSynced(header *replication.EventHeader, pos mysql.Position, set mysql.GTIDSet, force bool) error
+	// OnCreateUser is the create user event in QueryEvent
+	OnCreateUser(queryEvent *replication.QueryEvent, user *CreateUser) error
+	// OnDropUser is the drop user event in QueryEvent
+	OnDropUser(queryEvent *replication.QueryEvent, user *DropUser) error
+	// OnGrant is the grant event in QueryEvent
+	OnGrant(queryEvent *replication.QueryEvent, grant *Grant) error
+	// OnQueryEvent is query event (Note: Exclude OnCreateUser and OnDropUser and OnGrant)
```

Comment:
Maybe we should not add this method? Because this `EventHandler` does not try to cover all types of events. It will be more compatible that we only add well-defined callbacks. 

For example, `OnQueryEvent` may handle CreateIndexStmt, and if in future we decide to add a separate method for CreateIndexStmt, the usage of `OnQueryEvent` will be changed and user may meet problem without compile-time warning.

Reply:
I have re-adjusted the code and only added basic `queryEvent`. The detailed events inside (such as `CREATE USER`, `CREATE INDEX`, etc.) will be handled by the users themselves. our library provides basic event notification functionality."